arch/powerpc/platforms/pseries/setup.c

```diff
@@ -455,6 +455,139 @@ static void __init find_and_init_phbs(void)
 	of_pci_check_probe_only();
 }
 
+#ifdef CONFIG_PCI_IOV
+resource_size_t pseries_get_iov_fw_values(struct pci_dev *dev, int resno,
+			  int value)
+{
+	struct vf_bar_wdw {
+		__be64  addr;
+		__be64  aperture_size;
+		__be64  wdw_size;
+	};
+
+	struct vf_bar_wdw window_avail[PCI_SRIOV_NUM_BARS];
+	const int *indexes;
+	struct device_node *dn = pci_device_to_OF_node(dev);
+	int i, r, num_res;
+	resource_size_t return_value;
+
+	indexes = of_get_property(dn, "ibm,open-sriov-vf-bar-info", NULL);
+	if (indexes == NULL)
+		return  0;
+
+	memset(window_avail,
+	       0, sizeof(struct vf_bar_wdw)*PCI_SRIOV_NUM_BARS);
+	return_value = 0;
+	/*
+	 * First element in the array is the number of Bars
+	 * returned.  Search through the list to find the matching
+	 * bar
+	 */
+	num_res = of_read_number(&indexes[0], 1);
+	for (i = 2, r = 0; r < num_res
```

Comment:
Also all the static numbers in these functions, that isn't going to fly when upstreaming. They are going to want more defined names.

Reply:
Yep. thanks I should of given it an enum or constant. What do linux normally do? #define or some other enum?

Reply:
You can use either, usually enums are for a similar set of definitions.

So if you had like:

enum cmd_type {
then a list of different types of cmds }